packages/flutter_tools/bin/podhelper.rb

```diff
@@ -75,6 +75,7 @@ def flutter_additional_ios_build_settings(target)
     build_configuration.build_settings['ENABLE_BITCODE'] = 'NO'
 
     # Profile can't be derived from the CocoaPods build configuration. Use release framework (for linking only).
+    # TODO(stuartmorgan): Handle local engines here; see https://github.com/flutter/flutter/issues/132228
```

Comment:
Originally I was going to fix both, but iOS is messier and I don't know when I'll get back to it, so I decided to at least land the macOS part and document the pitfall before I lose track of the branch where I had fixed macOS.